Snowflake.Data/Core/HttpUtil.cs

```diff
@@ -404,8 +411,19 @@ protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage
                         }
                         else
                         {
-                            //TODO: Should probably check to see if the error is recoverable or transient.
-                            logger.Warn("Error occurred during request, retrying...", e);
+                            Exception innermostException = e;
+                            while (innermostException.InnerException != null) innermostException = innermostException.InnerException;

```

Comment:
I'm just thinking if we could fall into an infinitive loop if someone would throw an exception with itself assigned to inner exception. I know it would be weird but maybe we could add a condition for that in the while loop

Reply:
Added check for that case